src/poetry/installation/executor.py

```diff
@@ -729,9 +743,9 @@ def _populate_hashes_dict(self, archive: Path, package: Package) -> None:
     @staticmethod
     def _validate_archive_hash(archive: Path, package: Package) -> str:
         archive_hash: str = "sha256:" + get_file_hash(archive)
-        known_hashes = {f["hash"] for f in package.files}
+        known_hashes = {f["hash"] for f in package.files if f["file"] == archive.name}
 
-        if archive_hash not in known_hashes:
+        if known_hashes and archive_hash not in known_hashes:
```

Comment:
I am not sure about this. Is there a standard in pypy that a package downloaded from a url needs to provide a hash ? 

In case there is no hash we would run into an empty set here. On second though, it would already fail in line 746 accessing `f["hash"]`

I can remove it, if we can assume that every package.files entry has a hash

Reply:
We assumed it before this PR. If there the set of known hashes is empty, the hash is not known and we should fail for security reasons. I wouldn't weaken that constraint if it is not necessary.

Reply:
Done